agent/agent.go

```diff
@@ -94,6 +101,97 @@ type outputUnit struct {
 	outputs []*models.RunningOutput
 }
 
+func (a *Agent) RunSingleInput(input_config *models.InputConfig, ctx context.Context) {
```

Comment:
Probably want to return errors, so we know when something occurred successfully or not